plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-panel.php

```diff
@@ -37,21 +38,39 @@
 			</label>
 		<?php endforeach; ?>
 	</span>
+	
+	<?php if ( true ) : ?>
+		<dl class="product_data_cards wc-cards">
+			<?php foreach ( self::get_product_data_tabs() as $key => $tab_data ) : ?>
+				<dt class="clear <?php echo esc_attr( $key ); ?>_options_label <?php echo esc_attr( $key ); ?>_card_label <?php echo esc_attr( isset( $tab_data['class'] ) ? implode( ' ', (array) $tab_data['class'] ) : '' ); ?>">
+					<span><?php echo esc_html( $tab_data['label'] ); ?></span>
+				</dt>
+				<dd class="card <?php echo esc_attr( $key ); ?>_options <?php echo esc_attr( $key ); ?>_card <?php echo esc_attr( isset( $tab_data['class'] ) ? implode( ' ', (array) $tab_data['class'] ) : '' ); ?>">
+					<?php self::output_tab( $tab_data ); ?>
```

Comment:
By adding the single `output_tab()` method we can output the headers and their associated panels at the same time, which is nice.

The obvious problem with this version is that we aren't calling any hooks, but since they're just adding to the array it should be possible to remain compatible with existing implementations.

Reply:
In theory the hook system already provides for this via `$priority`, but it might be nice to let people be a little more intentional about it.

Reply:
> We might want to be careful using priority as a way to order since there are cases where we might want to hook into a later priority (e.g., to check if another tab has been registered) but still remain in a different position than the priority used.

I think I agree, but it's worth thinking hard about introducing parallel options. In this hypothetical, the lower-priority hook that wants to check for the existence of another tab does still have access to the whole tabs array, and can insert itself wherever it wants, yeah?